csharp/src/Google.Protobuf/MessageParser.cs

```diff
@@ -72,7 +72,6 @@ public IMessage ParseFrom(byte[] data)
         {
             IMessage message = factory();
             message.MergeFrom(data, DiscardUnknownFields, Extensions);
-            CheckMergedRequiredFields(message);
```

Comment:
what does this fix, is there a test that demonstrates the wrong behavior?

Reply:
I can write one. The wrong behavior can only be seen if you parse using the non-generic methods.

```cs
var foo = Foo.Parser.ParseFrom(data); // doesn't throw if missing required
foo = (Foo.Parser as MessageParser).ParseFrom(data); will throw if missing required
```

Reply:
Adding a test would be useful.